src/puya/models.py

```diff
@@ -45,22 +45,32 @@ class ARC4CreateOption(enum.Enum):
     disallow = enum.auto()
 
 
-def _freeze_list_of_lists(elements: Sequence[Sequence[str]]) -> Sequence[tuple[str, str]]:
-    return tuple((e1, e2) for (e1, e2) in elements)
-
-
 @attrs.frozen
-class ARC32StructDef:
+class ARC4StructField:
     name: str
-    elements: Sequence[tuple[str, str]] = attrs.field(default=(), converter=_freeze_list_of_lists)
+    type: str
+    struct: str | None
```

Comment:
What's this field for? 🤔

Reply:
ARC-56 is a backwards compatible extension of ARC-4, so `type` just defines the ARC4 encoding of an arg, which for a struct/named tuple will just be the tuple encoding with no field information. 
If the arg is actually a struct, then this field defines the name of the struct, which can be looked up on the Contract `structs` mapping to get the full information

See https://github.com/algorandfoundation/ARCs/pull/258/files#diff-6eb6fa5a0cc74e7fa2334541fb86699840a16e2518c98275de9268945c3dcaadR164 for more context